### PR TITLE
Torchfx dynamic shapes initial (CVS-139182)

### PR DESCRIPTION
**Pull request by cavusmustafa**
### Details:
 - Enabling dynamic shapes with aot_autograd

### Tickets:
 - CVS-139182

### CODE REVIEW

**Comment by cavusmustafa** [comment]
Please add label for code freeze

File: src/bindings/python/src/openvino/frontend/pytorch/fx_decoder.py [hunk 1]
```diff
@@ -56,6 +56,16 @@ def __init__(self, pt_module, fx_gm=None, nodes=None, mark_node_callback=None, i
                     uargs = self.unpack_containers(self._nodes[i].args)
                     self._outputs = [(arg[0], self._nodes.index(arg[1]))
                                      for arg in uargs if arg[1] is not None]
+            for idx, shape in enumerate(found_shapes):
+                new_shape=[]
```
Comment (p-wysocki) [inline src/bindings/python/src/openvino/frontend/pytorch/fx_decoder.py:60]:
```suggestion
```
This creation does not seem to be needed.

File: src/frontends/pytorch/src/op/expand.cpp [hunk 2]
```diff
@@ -42,24 +42,30 @@ OutputVector translate_expand_as(const NodeContext& context) {
 };
 
 OutputVector translate_expand_fx(const NodeContext& context) {
-    // aten::expand(Tensor(a) self, SymInt[] size, *, bool implicit=False) -> Tensor(a)
-    num_inputs_check(context, 2, 3);
+    auto num_inputs = context.get_input_size();
+    num_inputs_check(context, 2, num_inputs);
```
Comment (p-wysocki) [inline src/frontends/pytorch/src/op/expand.cpp:46]: Is this assertion still valid if we're getting `num_inputs` from the `context` itself?

File: src/frontends/pytorch/src/op/reshape.cpp [hunk 3]
```diff
@@ -22,6 +25,77 @@ OutputVector translate_reshape(const NodeContext& context) {
...
+                    auto unsqueeze =
+                        context.mark_node(std::make_shared<ov::op::v0::Unsqueeze>(context.get_input(0), zero));
+                    return {unsqueeze};
+                } else if (shape_vec.size() >= 3 && shape_vec.size() + 1 == inp_shape.size() && inp_shape[0] == 1 &&
+                           inp_shape[1] == shape_vec[0]) {
+                    // [1, N, ...] -> [N, ...] Can be translated to Squeeze
+                    auto squeeze = context.mark_node(std::make_shared<ov::op::v0::Squeeze>(context.get_input(0), zero));
+                    return {squeeze};
+                } else if (inp_shape.size() == shape_vec.size()) {
+                    // If the input rank is equal to output rank, we can use 0s in place of dynamic dymensions
```
Comment (p-wysocki) [inline src/frontends/pytorch/src/op/reshape.cpp:64]:
```suggestion
                    // If the input rank is equal to output rank, we can use 0s in place of dynamic dimensions
```

**Comment by p-wysocki** [comment]
Just minor things, not needed to be addressed unless other changes are requested.

File: src/frontends/pytorch/src/op/expand.cpp (hunk 2 above)
Comment (cavusmustafa) [reply src/frontends/pytorch/src/op/expand.cpp:46]: Checking the minimum number of inputs (which is at least 2). But probably it's very unlikely to be less then 2.

**Comment by suryasidd** [comment]
build_jenkins

**Review by suryasidd — APPROVED** [review]

**Pull request merged**